### ISSUE TITLE
PJAX - Duplicate Item on Back/Forward Button

### ISSUE THREAD

**Issue by tetrahydra** [snapappointments/bootstrap-select#1419]
Hello,

I am using PJAX for my site and clicking on back/forward button, the selectpicker will be shown as duplicates, more each time you click the button.

Is there any quick solution for this?

Thanks.

**Comment by ctf0** [comment]
had the same issue, use `$.pjax.defaults.maxCacheLength = 0;`

**Comment by tetrahydra** [comment]
Is it not going to affect the back/forward button? As the reason for using pjax is to take advantage on caching to allow back button to be utilise.

i could be wrong, by settings to 0, means that back/forward will need to be reloaded?

**Comment by ctf0** [comment]
have u tried it first ?

**Comment by nicomollet** [comment]
It works with `$.pjax.defaults.maxCacheLength = 0;` but Pjax cache has to be disabled in order to fix this bug.
My app would benefit from using the cache. I think we should a fix with pjax cache enabled.

**Comment by caseyjhol** [comment]
See [guidelines for contributing](https://github.com/silviomoreto/bootstrap-select/blob/master/CONTRIBUTING.md).

>## Bug reports
>
>A bug is a _demonstrable problem_ that is caused by the code in the repository.
>Good bug reports are extremely helpful - thank you!
>
>Guidelines for bug reports:
>
>1. **Use the GitHub issue search.** Check if the issue has already been reported.
>
>2. **Check if the issue has been fixed.** Try to reproduce it using the latest `master` or development branch in the repository.
>
>3. **Provide environment details.** Provide your operating system, browser(s), jQuery version, Bootstrap version, and bootstrap-select version.
>
>4. **Create an isolated and reproducible test case.** Create a [reduced test case](http://css-tricks.com/6263-reduced-test-cases/).
>
>5. **Include a live example.** Make use of jsFiddle or jsBin to share your isolated test cases.
